python/ray/_private/runtime_env/context.py

```diff
@@ -47,14 +51,14 @@ def deserialize(json_string):
         return RuntimeEnvContext(**json.loads(json_string))
 
     def exec_worker(self, passthrough_args: List[str], language: Language):
-        # TODO(Beacon): remove these when we're done      
+        # TODO(Beacon): remove these when we're done
         logger.debug(f"Worker context env: {self.env_vars}")
         update_envs(self.env_vars)
 
-        if language == Language.PYTHON and sys.platform == "win32":
-            executable = self.py_executable
-        elif language == Language.PYTHON:
-            executable = f"exec {self.py_executable}"
+        if language == Language.PYTHON:
+            executable = self.executable or sys.executable
+            if sys.platform == "win32":
+                executable = f"exec {executable}"
```

Comment:
same as above: I think it's a mistake to make `executable` control _both_ the julia and python executable that's used.